Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -74,6 +78,18 @@ class SerializerTest : public ::testing::Test {
     ExpectDeserializationRoundTrip(model, proto, type);
   }
 
+  /**
+   * Ensures that decoding fails with the given status.
+   *
+   * @param status the expected (failed) status. Only the code() is verified.
+   */
+  void ExpectFailedStatusDuringDecode(Status status,
```

Comment:
I'm not very familiar with GoogleTest, but won't using a helper function make it harder to read the stack trace in case the expectation is not fulfilled? If I'm not mistaken on this, then I think two features could be helpful: returning [AssertionResult](https://github.com/google/googletest/blob/master/googletest/docs/AdvancedGuide.md#using-a-function-that-returns-an-assertionresult) from this function (and then checking `EXPECT_TRUE(IsFailedStatus(...))`), or [SCOPED_TRACE](https://github.com/google/googletest/blob/master/googletest/docs/AdvancedGuide.md#adding-traces-to-assertions).